### PR TITLE
Add support in SourceMapConsumer for indexed source maps

### LINKED ISSUE #16: Support the "sections" alternate source map structure
https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k/edit?pli=1#heading=h.535es3xeprgt

### PR DESCRIPTION
**Pull request by jlfwong**
Fixes #16 (or at least the consumption part of it)

This is a WIP, but consumption of indexed source maps seems to be working. Let me know if you see anything going complete the wrong way, or if there are additional things that need implementing that I'm missing.

I'm not totally sure where the documentation for each of the functions should live - should I create stub functions on `SourceMapConsumer.prototype` that just throw errors saying they must be overridden, and place the documentation there? Or is it okay to duplicate the documentation between `BasicSourceMapConsumer` and `IndexedSourceMapConsumer`?

### CODE REVIEW

File: lib/source-map/indexed-source-map-consumer.js [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+   *    }],
+   *  }
+   *
+   * [0]: https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k/edit#heading=h.535es3xeprgt
+   */
+  function IndexedSourceMapConsumer(aSourceMap) {
+    var sourceMap = aSourceMap;
+    if (typeof aSourceMap === 'string') {
+      sourceMap = JSON.parse(aSourceMap.replace(/^\)\]\}'/, ''));
+    }
```
Comment (jlfwong) [inline lib/source-map/indexed-source-map-consumer.js]: I don't understand what this replacement is about, so I'm not sure what I'd name that function

File: lib/source-map/source-map-consumer.js [hunk 2]
```diff
@@ -184,241 +94,8 @@ define(function (require, exports, module) {
...
-          column: util.getArg(mapping, 'generatedColumn', null)
-        };
-      }
-
-      return {
-        line: null,
-        column: null
-      };
-    };
+      throw new Error("Subclasses must implement _parseMappings");
```
Comment (jlfwong) [inline lib/source-map/source-map-consumer.js]: `_generatedMappings` and `_originalMappings` are used by `eachMapping` (and possibly other places that I'm overlooking at the moment). Indexed source maps don't have mappings themselves, so the mappings that wind up listed here are the mappings of all of the submaps.

**Comment by jlfwong** [comment]
Doing a literal rebase proved crazy because of all the code flux since I opened this PR, so I just started from current master and ported my changes. I think I've addressed all the concerns except the ones I commented on (rebasing makes it hard to follow the comments though).

Unless there are significant blocking changes, I'd really appreciate merging this and dealing with minor issues in a later diff - keeping a diff this large up to date is tricky!

**Comment by fitzgen** [comment]
Thanks! I've merged the PR and published a new version to npm.
